From f40f133d6db8f882641a5375fd2ed112c8361df9 Mon Sep 17 00:00:00 2001
From: Torge Matthies <openglfreak@googlemail.com>
Date: Mon, 25 Mar 2024 09:19:18 +0100
Subject: [PATCH 08/53] winegstreamer: Do not create a read thread for
 uridecodebin-based media sources.

Rebased for wine-ge
CW-Bug-Id: #21644
---
 dlls/winegstreamer/media_source.c | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/dlls/winegstreamer/media_source.c b/dlls/winegstreamer/media_source.c
index c8dad61fd3c..6de0fdb5f4d 100644
--- a/dlls/winegstreamer/media_source.c
+++ b/dlls/winegstreamer/media_source.c
@@ -1661,9 +1661,12 @@ static HRESULT WINAPI media_source_Shutdown(IMFMediaSource *iface)
 
     wg_parser_disconnect(source->wg_parser);
 
-    source->read_thread_shutdown = true;
-    WaitForSingleObject(source->read_thread, INFINITE);
-    CloseHandle(source->read_thread);
+    if (source->read_thread)
+    {
+        source->read_thread_shutdown = true;
+        WaitForSingleObject(source->read_thread, INFINITE);
+        CloseHandle(source->read_thread);
+    }
 
     IMFMediaEventQueue_Shutdown(source->event_queue);
     IMFByteStream_Close(source->byte_stream);
@@ -1766,7 +1769,8 @@ static HRESULT media_source_create(struct object_context *context, IMFMediaSourc
     }
     object->wg_parser = parser;
 
-    object->read_thread = CreateThread(NULL, 0, read_thread, object, 0, NULL);
+    if (!context->url)
+        object->read_thread = CreateThread(NULL, 0, read_thread, object, 0, NULL);
 
     object->state = SOURCE_OPENING;
 
-- 
2.47.1